src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -1923,108 +1942,176 @@ export class WASMExpressionGen {
+                memberNameRef,
+                flag,
+            );
+            const propTypeIdRef = this.getPropTypeFromObj(
+                metaRef,
+                memberNameRef,
+                ItableFlag.UNKNOWN,
+            );
+            if (typeMeta.isInterface) {
+                return this.setInfcMember(
+                    typeMember,
+                    ownerType,
+                    thisRef,
+                    valueIdxInTypeMeta,
+                    targetValue,
+                    metaRef,
+                    indexRef,
+                    flagRef,
+                    propTypeIdRef,
+                );
+            } else {
+                return this.setObjMember(
+                    typeMember,
+                    thisRef,
+                    thisTypeRef,
+                    valueIdxInTypeMeta,
+                    targetValue,
+                );
+            }
         }
     }
 
-    private setInfcFieldWithSetter(
+    private setInfcMember(
         member: MemberDescription,
         infcType: ValueType,
         thisRef: binaryen.ExpressionRef,
-        fieldIdx: number,
+        valueIdx: number,
         targetValue: SemanticsValue,
+        metaRef: binaryen.ExpressionRef,
+        indexRef: binaryen.ExpressionRef,
+        flagRef: binaryen.ExpressionRef,
+        propTypeIdRef: binaryen.ExpressionRef,
     ) {
-        let res: binaryen.ExpressionRef;
-        const isSet = member.hasSetter ? false : true;
-        const callMethod = member.hasSetter ? true : false;
-        const targetValueRef = member.hasSetter
-            ? undefined
-            : this.wasmExprGen(targetValue);
-        const memberValueType = member.hasSetter
+        const propType = member.hasSetter
             ? (member.setter as VarValue).type
             : member.valueType;
-        res = this.getOriObjInfoByFindIdx(
+
+        const targetValueRef = this.wasmExprGen(targetValue);
+
+        /* TODO: workaround: quick path may fail, since cast failure */
+        const infcDescTypeRef = this.wasmTypeGen.getWASMObjOriType(infcType);
+        const castedObjRef = binaryenCAPI._BinaryenRefCast(
+            this.module.ptr,
             thisRef,
-            infcType,
-            member,
-            memberValueType,
-            fieldIdx,
-            isSet,
-            callMethod,
-            targetValueRef,
+            infcDescTypeRef,
         );
-        if (member.hasSetter) {
-            res = this.callFuncRef(
-                memberValueType,
-                res,
-                [targetValue],
-                thisRef,
+        const ifEqualTypeId = FunctionalFuncs.isTypeIdEqual(
+            this.module,
+            infcType.typeId,
+            metaRef,
+        );
+        let ifEqualTrue: binaryen.ExpressionRef;
+        if (propType.kind === ValueTypeKind.FUNCTION) {
+            /* if property's value type is function, and typeid is equal, then we can get property from vtable */
```

Comment:
"we can set property from vtable"?

Reply:
If we define a function in interface, then it must be recognized as a `method`, not a `field`, then it must be stored in vtable, and if interface's typeid is equal with class's typeid/impl_id, then in class, it must be a `method` too, we can get it from vtable, don't need to consider the situation that the function stored in instance as a field.

Reply:
I mean the function name is `setInfcProperty`, but the statement in the comment is "then we can get property ...", is it correct?

Reply:
Get it! I may copy and paste the comment from `getInfcProperty`, I will fix it.